src/librustdoc/clean/mod.rs

```diff
@@ -3503,7 +3634,11 @@ fn print_const_expr(cx: &DocContext, body: hir::BodyId) -> String {
 fn resolve_type(cx: &DocContext,
                 path: Path,
                 id: ast::NodeId) -> Type {
-    debug!("resolve_type({:?},{:?})", path, id);
+    if id == ast::DUMMY_NODE_ID {
```

Comment:
I'm wondering if it won't spawn a warning considering that `debug!` is removed in non-debug mode... Could you check about this please?

Reply:
I'm not quite sure what you mean. Do you think I should be unconditionally printing out dummy node ids?

Reply:
My point is that the `if else` will be useless if not compiled in debug mode.

Reply:
I've been compiling in release mode, and haven't gotten any warnings.